File: algorithms/interview-preparation-kit-1/php/code-response/spec/MinMaxSum/MinMaxSum.spec.php
<?php

use \MinMaxSum\MinMaxSum;

describe('Min Max Sum', function () {

  it('min and max of [1, 2, 3, 4, 5]', function () {
    $array = [1, 2, 3, 4, 5];
    $expected = [10, 14];
    expect(MinMaxSum::calc($array))->toEqual($expected);
  });

  it('min and max of [0, 2, 6, 3, 4]', function () {
    $array = [0, 2, 6, 3, 4];
    $expected = [9, 15];
    expect(MinMaxSum::calc($array))->toEqual($expected);
  });

  it('min and max of [10, 23, 61, 37, 41]', function () {
    $array = [10, 23, 61, 37, 41];
    $expected = [111, 162];
    expect(MinMaxSum::calc($array))->toEqual($expected);
  });

});

File: algorithms/interview-preparation-kit-1/min-max-sum/ruby/code/spec/min_max_sum.spec.rb
require_relative "../src/min_max_sum"

describe MinMaxSum do
  describe ".calc" do
    context "[1, 2, 3, 4, 5]" do
      it "return [10, 14]" do
        value = [1, 2, 3, 4, 5]
        expected = [10, 14]
        expect(MinMaxSum.calc(value)).to eql(expected)
      end
    end

    xcontext "[0, 2, 6, 3, 4]" do
      it "return [9, 15]" do
        value = [0, 2, 6, 3, 4]
        expected = [9, 15]
        expect(MinMaxSum.calc(value)).to eql(expected)
      end
    end

    xcontext "[10, 23, 61, 37, 41]" do
      it "return [111, 162]" do
        value = [10, 23, 61, 37, 41]
        expected = [111, 162]
        expect(MinMaxSum.calc(value)).to eql(expected)
      end
    end
  end
end
